app/controllers/state_file/questions/questions_controller.rb

```diff
@@ -56,8 +57,30 @@ def prev_step
         form_navigation.prev
       end
 
+      def prev_action
+        return unless self.class.navigation_actions.count > 1
+
+        current_action = action_name.to_sym
+
+        # look up action corresponding to the submitted form if we hit a validation error on submission and are re-rendering new/edit
+        replacements = { update: :edit, create: :new }
+        current_action = replacements[current_action] if replacements.key? current_action
+
+        action_index = self.class.navigation_actions.index(current_action)
+
+        case action_index
+        when 0 then nil
+        when 1.. then self.class.navigation_actions[action_index - 1]
```

Comment:
I think I'm ok with this as-is and as this is built upon current patterns of how we use `navigation_actions` in our app --

but just thoughts here...
wondering if this makes sense for all sorts of `navigation_actions` array, for example, if I made a `navigation_actions` that looks like `[:edit, :delete, :index]`, this flow wouldn't really work, right? 
Is more of what we're looking for that if we have more than one `navigation_actions` defined, we want to return to the `:index` action as a `prev_action`?

```
when 1.. then :index
```
or
```
when 1.. then self.class.navigation_actions[0]
```
with the understanding that if we're defining multiple actions on a controller and an index is specified, the first should always be an index?

Reply:
Good call - you're right, this code assumes both that:
- the `navigation_actions` will be a list that should be visited in order, and
- the first item in the `navigation_actions` should be some kind of "default" page that is visited before every other action in the list, and should be returned to if we can't figure out where else to send you.

That second assumption is already in the code in `StateFile::Questions::QuestionsController#path_for_step` where we say `options = { action: step.navigation_actions.first }`. 

However, the first assumption is new and really has no basis - it just happens to work right now because all the places where we've overridden `navigation_actions`, we've only included 2 actions. I'll change this code to jump back to the first action in the list if the current action is anything other than the first action, and to the previous controller if the current_action is the first action in the list. Does that sound good to you?